### DISCUSSION TITLE
Can not get third party extension setting options by PreferenceService

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-theia/theia#14078] [Q&A] [answered]
Hi,
   I install llvm-vs-code-extensions.vscode-clangd-0.1.29 in theia plugins, I want to get 'clangd.path' of its setting, I use follow code
```
     await this.preferenceService.ready;
    const exist = this.preferenceService.get('clangd.path');
```
![image](https://github.com/user-attachments/assets/9ad207e4-59b5-4fb1-af15-2309dde570aa)

 const exist = this.preferenceService.get('clangd.path'); always return undefine, even I have set value for 'clangd.path' from the UI setting. It is strange, Does I use correct method to get?

**Comment by msujew** [comment, marked as answer]
Hey @wss29,

I'm not sure waiting until the preference service is ready is enough - the backend registers the extension preferences with a bit of delay in the frontend. You probably need to wait until the `HostedPluginSupport#didStart` promise is ready:

```ts
@inject(HostedPluginSupport)
private readonly hostedPluginSupport: HostedPluginSupport;
...
await this.preferenceService.ready;
await this.hostedPluginSupport.didStart;
const exist = this.preferenceService.get('clangd.path');
```

**Reply by wss29** [reply]
Hi @msujew
 It works, thanks
